### ISSUE TITLE
Crash when dispose() is called on 0.1.2 and 0.1.3

### ISSUE THREAD

**Issue by Zarainia** [alexmercerind/dart_vlc#103]
Consider this code where the player is used in a dialog:

```dart
import 'package:dart_vlc/dart_vlc.dart';
import 'package:flutter/material.dart';

void main() {
  DartVLC.initialize();
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData.light(),
      home: Scaffold(
        appBar: AppBar(
          title: Text("Test"),
        ),
        body: Column(
          children: [AppButton()],
        ),
      ),
    );
  }
}

class AppButton extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialButton(onPressed: () {
      showDialog(context: context, builder: (context) => AlertDialog(content: SoundTest()));
    }, child: Text("Open"),);
  }
}

class SoundTest extends StatefulWidget {
  @override
  _SoundTestState createState() => _SoundTestState();
}

class _SoundTestState extends State<SoundTest> {
  late Player player;

  @override
  void dispose() {
    player.dispose();
    super.dispose();
  }

  @override
  void initState() {
    super.initState();
    player = Player(id: 1);
  }

  @override
  Widget build(BuildContext context) {
    return MaterialButton(onPressed: (){}, child: Text("Play"),);
  }
}
```

Calling `dispose()` on the `Player` (eg., to clean up when the dialog widget itself is disposed) causes the app to crash with no message. This did not use to happen before 0.1.1.

**Comment by alexmercerind** [comment]
First of all, big thanks for giving a running reproducible example. Very rare to see in non-moderated issues.

This issue is fixed in #104.

Will close this, once merged.

**Comment by alexmercerind** [comment]
Fixed on master. Closing.
